tests/functional/v2/cache/testRedisBackend.js

```diff
@@ -0,0 +1,90 @@
+const assert = require('assert');
+const uuid = require('uuid');
+const { RedisCache } = require('../../../../libV2/cache').backends;
+const { shardFromTimestamp } = require('../../../../libV2/utils');
+const schema = require('../../../../libV2/cache/schema');
+const { generateFakeEvents } = require('../../../utils/v2Data');
+
+const startTime = new Date().getTime();
+const endTime = startTime + 30000; // Add 30 seconds;
+const testValues = generateFakeEvents(startTime, endTime, 50);
+
+describe('Test cache redis backend', () => {
+    let cache;
+    let prefix;
+    beforeEach(done => {
+        prefix = uuid.v4();
+        cache = new RedisCache({}, prefix);
+        cache.connect().then(() => done());
+    });
+
+    it('should create a new redis backend instance', () => {
+        assert.equal(cache instanceof RedisCache, true);
+    });
+
+    testValues.map(event =>
+        it('should set and get a key', () => {
+            const key = schema.getUtapiMetricKey('test', event);
+            return cache
+                .setKey(key, JSON.stringify(event.getValue()))
+                .then(res => {
+                    assert.strictEqual(true, res);
+                    return cache.getKey(key);
+                })
+                .then(value => {
+                    assert.deepStrictEqual(JSON.parse(value), event.getValue());
+                });
+        }));
+
+    testValues.map(event =>
+        it('should add a key to a shard', () => {
+            const shard = shardFromTimestamp(event.timestamp);
+            return cache.addToShard(shard, event).then(res => {
+                assert.strictEqual(true, res);
+            });
+        }));
+
+    it('should fetch the keys in a shard', () => {
+        const shard = shardFromTimestamp(new Date().getTime());
+        const expected = testValues.map(e =>
+            schema.getUtapiMetricKey(prefix, e));
+        return Promise.all(
+            testValues.map(event => cache.addToShard(shard, event)),
+        )
+            .then(() => cache.getKeysInShard(shard))
+            .then(keys => {
+                assert.deepStrictEqual(keys.sort(), expected.sort());
+            });
+    });
```

Comment:
I just want to demo what this would look like using async + await, I know it is still up to discussion but this is a good example. Note I have not run the tests so a couple of tweaks might be in order but if serious thought is taken on converting it should not be too hard.

```suggestion
    it('should fetch the keys in a shard', async () => {
        const shard = shardFromTimestamp(new Date().getTime());
        const expected = testValues.map(e =>
            schema.getUtapiMetricKey(prefix, e)
        );
        await Promise.all(testValues.map(event => cache.addToShard(shard, event)));
        let keys = await cache.getKeysInShard(shard);
        assert.deepStrictEqual(keys.sort(), expected.sort());
    });
```

cc @rahulreddy @anurag4DSB

Reply:
I'll double check but I believe passing an `async` function as a callback is a linting error

Reply:
I was wrong, thinking of this rule https://eslint.org/docs/rules/no-async-promise-executor